presto-main/src/main/java/com/facebook/presto/execution/SetSessionTask.java

```diff
@@ -84,6 +85,12 @@ public String getName()
 
         String value = serializeSessionProperty(type, objectValue);
 
+        if ((propertyName.toString().equals(TASK_WRITER_COUNT)) && (Integer.bitCount(Integer.parseInt(value)) != 1)) {
```

Comment:
SetSessionPropertyTask should not contain any property-specific checks or validations. This should be done in the validator for the task writer count property (which it already does). So, the question we need to answer is why do the validations not work in this context.